Clients/GUI/Pages/SignUp.razor

```diff
@@ -1,57 +1,62 @@
-                <label style=“color:red;display:@isErrorDisplayed”> @errorMessage </label>
-            </div>
-            <br />
-            <div class="form-group">
-                <label for="Password">Password</label>
-                <InputText class="form-control col-3" type="password" @bind-Value="userInput.Password" placeholder="*****" />
-            </div>
-            <br />
-            <div class="form-group">
-                <label for="ConfirmPassword">Confirm Password</label>
-                <InputText class="form-control col-3" type="password" @bind-Value="userInput.ConfirmPassword" placeholder="*****" />
-            </div>
-            <br />
-            <ValidationSummary />
-            <br />
-
-            <input type="submit" class="form-control col-1 button btn-primary" value="Sign up" />
-
-        </EditForm>
-    </div>
+<div class="col-12">
+    <h3><b>Sign up</b></h3>
+    <hr />
+
+    <EditForm Model="@userInput" OnValidSubmit="@TrySignUp">
+        <DataAnnotationsValidator />
+        <div class="form-group">
+            <label for="Email">Email</label>
+            <InputText class="form-control col-3" @bind-Value="userInput.Email" placeholder="example@gmail.com" />
+            <label style=“color:red;display:@isErrorDisplayed”> @errorMessage </label>
+        </div>
+        <br />
+        <div class="form-group">
+            <label for="Password">Password</label>
+            <InputText class="form-control col-3" type="password" @bind-Value="userInput.Password" placeholder="*****" />
+        </div>
+        <br />
+        <div class="form-group">
+            <label for="ConfirmPassword">Confirm Password</label>
+            <InputText class="form-control col-3" type="password" @bind-Value="userInput.ConfirmPassword" placeholder="*****" />
+        </div>
+        <br />
+        <ValidationSummary />
+        <br />
+
+        <input type="submit" class="form-control col-1 button btn-primary" value="Sign up" />
+
+    </EditForm>
+</div>
 
 
 @code {
     bool isErrorDisplayed = false;
     string errorMessage;
     private SignUpViewModel userInput;
 
-    public void TrySignUp()
+    public async Task TrySignUp()
     {
         try
         {
-            SignUpper.SignUp(new UserEmailPassword(userInput.Email, userInput.Password));
-            NavigationManager.NavigateTo("/signin");
+            var hash = ShaHash.GetHash(userInput.Password);
+            var credentials = new UserEmailPassword(userInput.Email, hash);
+            SignUpper.SignUp(credentials);
+            await LocalStorage.ClearAsync();
+            var userToken = SignInner.SignIn(credentials);
+            LocalStorage.SetItemAsync(userToken.UserId.ToString(), userToken.Body);
+            AuthenticationProvider.MarkSignedIn();
+            NavigationManager.NavigateTo("/userinfo");
         }
         catch
```

Comment:
Don't we suppose to catch customs exceptions? and have different handling for them?

Reply:
Because our custom exceptions need to be updated (check new pr)

Reply:
Fixing it in next commit